### PR TITLE
Issue 103: HouseController unit tests

### LINKED ISSUE #103: HouseController unit tests
## Task description

There is a need to create unit tests for `HouseController`.
As an example, you can have a look at `UserControllerTest`.

## Task Relationships
This task is a sub-task of: #100

### PR DESCRIPTION
**Pull request by DDmit04**
## 🚀 Description

add unit tests for `HouseController `

## 📄 Motivation and Context

Closes #103

### CODE REVIEW

**Comment by DDmit04** [comment]
srange,on my PC with IDEA tests are passed (`gradlew build` passed too)

**Comment by DDmit04** [comment]
For some reasons this lines fails test
```
verify(communityRepository).findByCommunityId(TEST_COMMUNITY_ID);
testCommunityHouses.forEach(house -> verify(communityHouseRepository).deleteByHouseId(house.getHouseId()));
```

**Comment by DDmit04** [comment]
is this necessary to use `@Transactional` on this methods?
```
@Transactional
  public boolean deleteCommunity(String communityId)
....
@Transactional
  public boolean removeHouseFromCommunityByHouseId(String communityId, String houseId)
```

File: src/main/java/com/myhome/controllers/HouseController.java [hunk 1]
```diff
@@ -72,27 +75,31 @@
...
-    CommunityHouse houseDetail =
-        houseService.getHouseDetailsById(houseId).get();
-    GetHouseDetailsResponse.CommunityHouse getHouseDetailsResponse =
-        houseApiMapper.communityHouseToRestApiResponseCommunityHouse(houseDetail);
-
-    GetHouseDetailsResponse response = new GetHouseDetailsResponse();
-    response.getHouses().add(getHouseDetailsResponse);
-    return ResponseEntity.status(HttpStatus.OK).body(response);
+    Optional<CommunityHouse> communityHouseOptional = houseService.getHouseDetailsById(houseId);
+    return communityHouseOptional
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/HouseController.java]: Can we simplify this to:
```
return houseService.getHouseDetailsById(houseId)
   .map<rest of the code>
```

File: src/main/java/com/myhome/controllers/HouseController.java [hunk 2]
```diff
@@ -72,27 +75,31 @@
...
+        .map(houseApiMapper::communityHouseToRestApiResponseCommunityHouse)
+        .map(Collections::singleton)
+        .map(GetHouseDetailsResponse::new)
+        .map(ResponseEntity::ok)
+        .orElse(ResponseEntity.status(HttpStatus.NOT_FOUND).body(new GetHouseDetailsResponse()));
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/HouseController.java]:
```
        .orElse(ResponseEntity.notFound().build());
```

File: src/test/java/com/myhome/controllers/unit/HouseControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (mslowiak) [inline src/test/java/com/myhome/controllers/unit/HouseControllerTest.java]: wildcard import

File: src/test/java/com/myhome/controllers/unit/HouseControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+  @BeforeEach
+  private void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  private String generateUniqueId() {
+    return UUID.randomUUID().toString();
+  }
+
+  private Set<CommunityHouse> getTestHouses(int count) {
```
Comment (mslowiak) [inline src/test/java/com/myhome/controllers/unit/HouseControllerTest.java]: Please move helper methods to the bottom (tests first)

File: src/test/java/com/myhome/services/unit/CommunitySDJpaServiceTest.java [hunk 5]
```diff
@@ -343,10 +334,7 @@ void deleteCommunity() {
 
     // then
     assertTrue(communityDeleted);
-    verify(communityRepository).findByCommunityId(TEST_COMMUNITY_ID);
-    testCommunityHouses.forEach(house -> {
-      verify(communityHouseRepository).deleteByHouseId(house.getHouseId());
-    });
+    verify(communityRepository ,times(TEST_HOUSES_COUNT + 1)).findByCommunityId(TEST_COMMUNITY_ID);
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/unit/CommunitySDJpaServiceTest.java]: Please apply the correct formatting

**Review by mslowiak — CHANGES_REQUESTED** [review]

**Review by mslowiak — APPROVED** [review]

**Pull request merged**